### PR TITLE
Out of range error and sizeof(PointT)

### PR DESCRIPTION
**Pull request by xmfcx**
Replace point.step with sizeof(PointT)

Use out_of_range exception

### CODE REVIEW

File: sensor_msgs/include/sensor_msgs/point_cloud_msg_wrapper.hpp [hunk 1]
```diff
@@ -265,31 +265,35 @@ class PointCloudMsgWrapper
...
   }
   /// Get a point reference at the specified index. Only compiled if message is not const.
   /// @throws std::runtime_error if the index is out of bounds.
   COMPILE_IF_MUTABLE(CloudMsgT, PointT &) at(const std::size_t index)
   {
     if (index >= size()) {
-      throw std::runtime_error("Cannot get point, index too large.");
+      throw std::out_of_range(
+              "Index is out of bounds, " +
+              std::to_string(index) + " >= " + std::to_string(size()));
```
Comment (niosus) [inline sensor_msgs/include/sensor_msgs/point_cloud_msg_wrapper.hpp:282]: :clap: I find this message to be much nicer. Thanks!

**Review by niosus — APPROVED** [review]
Thanks! I like these changes. Just one thing about the commit message. Ros2 has a policy on commits that they must be signed-off by the user (check out my commits in my PR to ros2 here: https://github.com/ros2/common_interfaces/pull/141). If you sign your commits just like that and we are careful they might survive it until merged into the ros2 master.

**Comment by xmfcx** [comment]
> Thanks! I like these changes. Just one thing about the commit message. Ros2 has a policy on commits that they must be signed-off by the user (check out my commits in my PR to ros2 here: [ros2#141](https://github.com/ros2/common_interfaces/pull/141)). If you sign your commits just like that and we are careful they might survive it until merged into the ros2 master.

Ok signed in the commit message and force pushed it.

**Comment by niosus** [comment]
@xmfcx merged. Let's see if the ros2 CI is happy about it :wink: 

I will ping you on the PR at the points I'd like to talk more about/improve if that's fine with you.

**Comment by xmfcx** [comment]
> I will ping you on the PR at the points I'd like to talk more about/improve if that's fine with you.

Sure, I'd be happy to help anytime, thanks.

**Pull request merged**